File: front/Сonsumer/src/interface/DTO/chat/ChatListItemDto.ts
import type { ChatUserDto } from "./ChatUserDto";
import type { LastMessagePreviewDto } from "./LastMessagePreviewDto";

export interface ChatListItemDto {
  id: number;
  name: string | null;
  type: string;
  isPublic: boolean;
  isMember: boolean;
  myRole: string | null;
  avatarUrl: string | null;
  avatarIsVideo: boolean;
  companion: ChatUserDto | null;
  lastMessage: LastMessagePreviewDto | null;
  createdAt: string;
}

File: back/ChatService.Application/DTO/ChatListItemDto.cs
namespace ChatService.Application.DTO
{
    public class ChatListItemDto
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string Type { get; set; } = string.Empty;
        public bool IsPublic { get; set; }
        public bool IsMember { get; set; }
        public string? MyRole { get; set; }
        public string? AvatarUrl { get; set; }

        public bool AvatarIsVideo { get; set; }
        public ChatUserDto? Companion { get; set; }
        public LastMessagePreviewDto? LastMessage { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}
